incorrect contribution guide instructions

## Context
In addition to #1785, I discovered that the instructions for ["Validating the data"](https://github.com/mdn/browser-compat-data/blob/master/CONTRIBUTING.md#validating-the-data) in the contribution guide misinforms contributors for node dependency installation.

### Current Instructions (devDependencies only)
The guide instructs you to run `npm install --only=dev` to install dependencies to run `npm test`. However, the tests rely on the `extend` dependency (which is not a `devDependency`).

As a result, a contributor would see something akin to the following:
```bash
$ npm install --only=dev

> mdn-confluence@0.0.7 postinstall /path/to/browser-compat-data/node_modules/mdn-confluence
> mkdir -p data/browser-compat-data && npm run build


> mdn-confluence@0.0.7 build /path/to/browser-compat-data/node_modules/mdn-confluence
> node ./scripts/build.js

added 232 packages in 7.845s
```

```bash
$ npm test

> mdn-browser-compat-data@0.0.46 test /path/to/browser-compat-data
> npm run lint


> mdn-browser-compat-data@0.0.46 lint /path/to/browser-compat-data
> node test/lint

module.js:549
    throw err;
    ^

Error: Cannot find module 'extend'
    at Function.Module._resolveFilename (module.js:547:15)
    at Function.Module._load (module.js:474:25)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/path/to/browser-compat-data/index.js:4:16)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! mdn-browser-compat-data@0.0.46 lint: `node test/lint`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the mdn-browser-compat-data@0.0.46 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     ~/.npm/_logs/2018-08-10T19_45_47_335Z-debug.log
npm ERR! Test failed.  See above for more details.
```

## Working Scenario (full npm install)
If I run a full install of node modules (`npm install`). This is what I a contributor might see:

```bash
$ npm install
npm WARN mdn-confluence@0.0.7 requires a peer of mdn-browser-compat-data@git://github.com/mdn/browser-compat-data.git but none is installed. You must install peer dependencies yourself.

added 1 package in 3.916s
```

```bash
$ npm test

> mdn-browser-compat-data@0.0.46 test /path/to/browser-compat-data
> npm run lint


> mdn-browser-compat-data@0.0.46 lint /path/to/browser-compat-data
> node test/lint

api/ANGLE_instanced_arrays.json
  JSON schema – OK
  Style – OK
  Browser versions – OK
api/AbortController.json
  JSON schema – OK
  Style – OK
  Browser versions – OK
...
```


## Questions
Should `extend` be a `devDependency` or are the contribution guidelines incorrect?

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -14,7 +14,7 @@ Let us know if these look good to you.
 * **Publishing a new package (staff only):** A new npm package should be published regularly, [see below](#publishing-a-new-package-version) for details.
 
 ## Validating the data
-You can use `npm test` to validate data against the schema. You might need to install the devDependencies using `npm install --only=dev`.
+You can use `npm test` to validate data against the schema. You might need to install the devDependencies using `npm install`.
 The JSON data is validated against the schema using [`ajv`](http://epoberezkin.github.io/ajv/).
 
 ### Optional: Validate/cross-reference against Web API Confluence Dashboard

